pkg/controller/servicediscovery/commonnamespace/hooks.go

```diff
@@ -129,12 +130,18 @@ func (h *Hooks) Observe(ctx context.Context, mg cpresource.Managed) (managed.Ext
 		return managed.ExternalObservation{},
 			awsclient.Wrap(cpresource.Ignore(ActualIsNotFound, err), errGetNamespace)
 	}
+
 	cr.SetConditions(xpv1.Available())
+
 	lateInited := false
 	if awsclient.StringValue(cr.GetDescription()) == "" {
-		cr.SetDescription(nsReqResp.Namespace.Description)
-		lateInited = true
+		if awsclient.StringValue(nsReqResp.Namespace.Description) != "" {
```

Comment:
```suggestion
		if awsclient.StringValue(cr.GetDescription()) == "" && awsclient.StringValue(nsReqResp.Namespace.Description) != "" {
```

Reply:
@muvaf  done and squashed commits - thanks for review